src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -50,30 +69,89 @@ export default class SqlX extends SelfServeBaseClass {
+      }
+    }
+
+    const enableDedicatedGateway = currentValues.get("enableDedicatedGateway")?.value as boolean;
+
+    if (response.status !== undefined) {
+      if (!enableDedicatedGateway) {
+        try {
+          await deleteDedicatedGatewayResource();
+          return { message: "DedicatedGateway resource will be deleted.", type: SelfServeNotificationType.info };
+        } catch (e) {
+          return {
+            message: "Deleting Dedicated Gateway resource failed. DedicatedGateway will not be deleted.",
+            type: SelfServeNotificationType.error,
+          };
+        }
+      } else {
+        // Check for scaling up/down/in/out
+      }
+    } else {
+      if (enableDedicatedGateway) {
+        const sku = currentValues.get("sku")?.value as string;
+        const instances = currentValues.get("instances").value as number;
+        try {
+          await updateDedicatedGatewayResource(sku, instances);
+          return { message: "Dedicated Gateway resource will be provisioned.", type: SelfServeNotificationType.info };
+        } catch (e) {
+          return {
+            message: "Updating Dedicated Gateway resource failed. Dedicated Gateway will not be updated.",
+            type: SelfServeNotificationType.error,
+          };
+        }
+      }
+    }
+    return { message: "No updates were applied at this time", type: SelfServeNotificationType.warning };
   };
 
   public initialize = async (): Promise<Map<string, SmartUiInput>> => {
-    // TODO: get initialization data from initializeDedicatedGatewayProvisioning() RP call.
-    throw new Error("onSave not implemented");
+    // TODO: RP call to check if dedicated gateway has already been provisioned.
+    // Based on the RP call enableDedicatedGateway will be true if it has not yet been enabled and false if it has.
+    const defaults = new Map<string, SmartUiInput>();
+    const enableDedicatedGateway = false;
+    defaults.set("enableDedicatedGateway", { value: enableDedicatedGateway, hidden: false, disabled: false });
+    defaults.set("sku", { value: "Cosmos.D4s", hidden: !enableDedicatedGateway, disabled: false });
+    defaults.set("instances", { value: await getInstancesMin(), hidden: !enableDedicatedGateway, disabled: false });
+    const response = await getCurrentProvisioningState();
+    if (response.status !== undefined) {
+      disableAttributesOnDedicatedGatewayChange = true;
+      defaults.set("enableDedicatedGateway", { value: true, hidden: false, disabled: false });
+      defaults.set("sku", { value: response.sku, hidden: false, disabled: true });
+      defaults.set("instances", { value: response.instances, hidden: false, disabled: true });
+    }
+
+    return defaults;
   };
 
   @Values({
     description: {
-      textTKey: "Provisioning dedicated gateways for SqlX accounts.",
+      textTKey: "DedicatedGatewayDescription",
       link: {
         href: "https://docs.microsoft.com/en-us/azure/cosmos-db/introduction",
```

Comment:
this should point to a sqlx specific documentation right?

Reply:
https://msdata.visualstudio.com/CosmosDB/_workitems/edit/1048189